src/main/java/hudson/tasks/Mailer.java

```diff
@@ -362,16 +370,35 @@ private static Session createSession(String smtpHost, String smtpPort, boolean u
             	 * and thats done in mail sender, and it would be a bit of a hack to get it all to
             	 * coordinate, and we can make it work through setting mail.smtp properties.
             	 */
-            	if (props.getProperty("mail.smtp.socketFactory.port") == null) {
+            	if (props.getProperty(SMTP_SOCKETFACTORY_PORT_PROPERTY) == null) {
                     String port = smtpPort==null?"465":smtpPort;
-                    props.put("mail.smtp.port", port);
-                    props.put("mail.smtp.socketFactory.port", port);
+                    props.put(SMTP_PORT_PROPERTY, port);
+                    props.put(SMTP_SOCKETFACTORY_PORT_PROPERTY, port);
             	}
             	if (props.getProperty("mail.smtp.socketFactory.class") == null) {
-            		props.put("mail.smtp.socketFactory.class","javax.net.ssl.SSLSocketFactory");
+                    // TODO SocketFactory properties are now discouraged. Using (`mail.smtp.ssl.enable`, "true") should suffice
+                    // https://javaee.github.io/javamail/FAQ#commonmistakes
+            		// Example also found at https://javaee.github.io/javamail/FAQ#smtpssl
+                    props.put("mail.smtp.socketFactory.class","javax.net.ssl.SSLSocketFactory");
```

Comment:
Looks like there's some weird indenting going on here.

I think the old code is using tabs, and your using spaces.